enclave-runner/src/usercalls/mod.rs

```diff
@@ -310,26 +310,32 @@ impl EnclaveState {
     fn new(
         kind: EnclaveKind,
         event_queues: FnvHashMap<TcsAddress, Mutex<Sender<u8>>>,
-    ) -> Arc<Self> {
+        user_call_ext_impl: Option<Box<UsercallExtension>>) -> Arc<Self> {
         let mut fds = FnvHashMap::default();
         fds.insert(FD_STDIN, Arc::new(FileDesc::stream(Shared(io::stdin()))));
         fds.insert(FD_STDOUT, Arc::new(FileDesc::stream(Shared(io::stdout()))));
         fds.insert(FD_STDERR, Arc::new(FileDesc::stream(Shared(io::stderr()))));
         let last_fd = AtomicUsize::new(fds.keys().cloned().max().unwrap() as _);
 
+        let user_call_ext_impl = user_call_ext_impl.unwrap_or_else( || {
```

Comment:
Don't need `unwrap_or_else` here, Box doesn't allocate for zero-sized types.
```rust
user_call_ext_impl.unwrap_or(Box::new(ConnectStreamDefault))
```
or even (after you make the impl suggested elsewehere)
```rust
user_call_ext_impl.unwrap_or(ConnectStreamDefault.into())
```